components/util.py

```diff
@@ -64,45 +66,80 @@ def build_menu(
 
 
 def rate_limit_tracker(_: Update, context: CallbackContext) -> None:
-    data = cast(Dict, context.chat_data).setdefault('rate_limit', {})
+    data = cast(Dict, context.chat_data).setdefault("rate_limit", {})
 
     for key in data.keys():
         data[key] += 1
 
 
-Func = TypeVar('Func', bound=Callable[[Update, CallbackContext], None])
+Func = TypeVar("Func", bound=Callable[[Update, CallbackContext], None])
 
 
 def rate_limit(
     func: Callable[[Update, CallbackContext], None]
 ) -> Callable[[Update, CallbackContext], None]:
     """
     Rate limit command so that RATE_LIMIT_SPACING non-command messages are
-    required between invocations.
+    required between invocations. Private chats are not rate limited.
     """
 
     @wraps(func)
     def wrapper(update: Update, context: CallbackContext) -> None:
+        if chat := update.effective_chat:
+            if chat.type == chat.PRIVATE:
+                return func(update, context)
```

Comment:
```suggestion
        if update.effective_chat.type == Chat.PRIVATE:
            return func(update, context)
```
`chat` isn't being used anywhere in that func

Reply:
basically this: https://github.com/python-telegram-bot/rules-bot/pull/78#discussion_r712488767

mypy doesn't know that `update.effective_chat is not None`

Reply:
oh man do we really need mypy since we actually can't really 'test' this code?

Reply:
No - code checkers are purely optional. I've just grown accustomed to using it and it did help me find some bugs in e.g. the search functionality. There is the [`strict_optional`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-strict_optional) flag, but I haven't figured out how to set this only for the `telegram` package …
My workflow usually is: write code without `cast`, run mypy, read the errors and then think about which of the `Optional` errors are actually valid and which can be ignored via `cast`. ofc this is not apparent when reading the code :D